core/base/morseSmaleQuadrangulation/MorseSmaleQuadrangulation.h

```diff
@@ -49,25 +58,26 @@ namespace ttk {
     }
 
     inline void setDualQuadrangulation(const bool input) {
-      dualQuadrangulation_ = input;
+      DualQuadrangulation = input;
     }
     inline void setShowResError(const bool value) {
-      showResError_ = value;
+      ShowResError = value;
     }
     inline void setInputPoints(const void *const addr) {
       inputPoints_ = static_cast<const float *>(addr);
     }
-    inline void setupTriangulation(Triangulation *const triangl) {
-      triangulation_ = triangl;
-      if(triangulation_ != nullptr) {
-        triangulation_->preconditionVertexNeighbors();
-        triangulation_->preconditionVertexTriangles();
-        triangulation_->preconditionBoundaryVertices();
+    inline void
+      preconditionTriangulation(AbstractTriangulation *const triangl) {
+      if(triangl != nullptr) {
+        triangl->preconditionVertexNeighbors();
+        triangl->preconditionVertexTriangles();
+        triangl->preconditionBoundaryVertices();
       }
-      verticesNumber_ = triangulation_->getNumberOfVertices();
+      verticesNumber_ = triangl->getNumberOfVertices();
     }
 
-    int execute();
+    template <typename triangulationType>
+    int execute(triangulationType &triangulation);
```

Comment:
quick note: for all "execute"-type functions, I'd suggest to use ```const``` input triangulations whenever possible.

Reply:
Addressed in 1a46b1f